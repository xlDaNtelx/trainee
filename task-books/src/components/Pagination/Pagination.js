import React from 'react';
import PropTypes from 'prop-types';

class Pagination extends React.Component {
  constructor(props) {
    super(props);
    this.nextPage = this.nextPage.bind(this);
    this.prevPage = this.prevPage.bind(this);
  }

  /**
  * Generate pagination.
  *
  * @public
  */
  setPagination() {
    const { count, perPage, page } = this.props;
    const pageCount = Math.ceil(count / perPage);
    return (
      <div className="row controls">
        <button type="button" className="prev small" disabled={page === 1} onClick={this.prevPage}>
Prev
        </button>
        <button type="button" className="next" disabled={!(page < pageCount)} onClick={this.nextPage}>
Next
        </button>
      </div>
    );
  }

  /**
  * Change current page to the next one.
  *
  * @public
  */
  nextPage() {
    const { onChange, page, perPage } = this.props;
    onChange(page + 1, perPage);
  }

  /**
  * Change current page to the previously one.
  *
  * @public
  */
  prevPage() {
    const { onChange, page, perPage } = this.props;
    onChange(page - 1, perPage);
  }

  render() {
    return (
      <div>
        {this.setPagination()}
      </div>
    );
  }
}

Pagination.propTypes = {
  /** Current page */
  page: PropTypes.number.isRequired,
  /** Count of page to display per page */
  perPage: PropTypes.number.isRequired,
  /** Function that will be call after using pagination */
  onChange: PropTypes.func.isRequired,
  /** Count of all items */
  count: PropTypes.number.isRequired,
};

export default Pagination;
