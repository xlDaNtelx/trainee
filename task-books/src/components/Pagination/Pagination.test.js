import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Pagination from './index';

describe('Pagination', () => {
  it('renders with all params', () => {
    const component = renderer.create(
      <Pagination page={1} perPage={0.5} count={8} onChange={() => { }} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders without some param', () => {
    const component = renderer.create(
      <Pagination page={1} perPage={0.5} count={8} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

test('Pagination functions test', () => {
  const pagination = shallow(<Pagination page={1} perPage={0.5} count={8} onChange={() => { }} />);
  pagination.find('.next').simulate('click');
  pagination.find('.prev').simulate('click');
});
