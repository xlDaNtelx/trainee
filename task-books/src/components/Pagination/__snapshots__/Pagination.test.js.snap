// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination renders with all params 1`] = `
<div>
  <div
    className="row controls"
  >
    <button
      className="prev small"
      disabled={true}
      onClick={[Function]}
      type="button"
    >
      Prev
    </button>
    <button
      className="next"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Next
    </button>
  </div>
</div>
`;

exports[`Pagination renders without some param 1`] = `
<div>
  <div
    className="row controls"
  >
    <button
      className="prev small"
      disabled={true}
      onClick={[Function]}
      type="button"
    >
      Prev
    </button>
    <button
      className="next"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Next
    </button>
  </div>
</div>
`;
